### DISCUSSION TITLE
Search through backups more easily

### DISCUSSION THREAD

**Discussion by christfried** [borgbase/vorta#1281] [Ideas]
I don't know if this is the right place but I miss a simple way to search through old backups. The mounting process is kind of complicated to do, especially for non-technical people. My father at least would not know what to do ...

Right now you need to:
1. select an archive
2. use mount from the drop-down menu
3. look through the whole tree to find a suitable folder or create one first (where no write protection is in place, so only home folder probably), select it, hit OK
4. open your file manager at the exact position you just had to browse to
5. afterwards you even need to manually unmount the archive

When I was trying out [Pika Backup](https://apps.gnome.org/de/app/org.gnome.World.PikaBackup/) it was much more simple than that. You could just click on the archive you need and open its contents inside a new file browser window. Close it again and be done. I liked the experience a lot. (I like Vorta for all of its features more of course ... ;) Would it be possible to implement that here, too?

Thanks for considering!

**Comment by m3nu** [comment]
You can mount the whole repo at once with the latest release.

Opening the mount path or managing the mount folder would be possible. That would make it single-click. But I wouldn't want it as default action.

**Reply by real-yfprojects** [reply]
> managing the mount folder would be possible

What do you mean by that?

**Reply by m3nu** [reply]
At the moment you have to 

1. Select the archive
2. Click "Mount"
3. Select the folder to mount to
4. Wait a bit
5. Open the selected folder

Steps 3 and 5 could be skipped by using some temporary folder created by Vorta.

**Reply by real-yfprojects** [reply]
This could be called "Fast/Quick mount" in the UI.

**Reply by christfried** [reply]
> But I wouldn't want it as default action.

Can you explain why you think so?

> Steps 3 and 5 could be skipped by using some temporary folder created by Vorta.

Sounds great! :)

**Reply by coelner** [reply]
maybe I'm missing something, but choosing an archive means that you mount this specific archive,  maybe the mount option `versions` could be placed under the repository tab to mount a merged view. 

the problem is not really solved, I would say it is a workaround:
https://github.com/borgbackup/borg/issues/4092

they suggest that you generate a list and grep on that to imitate a `locate` feature.

**Reply by m3nu** [reply]
Borgmatic recently added this feature. Can copy from them.

https://twitter.com/borgmatic/status/1533662986862092288

**Reply by real-yfprojects** [reply]
> maybe I'm missing something, but choosing an archive means that you mount this specific archive, maybe the mount option `versions` could be placed under the repository tab to mount a merged view.

In _vorta_ you can also mount all archives at once as subfolders of the mount location.

> the problem is not really solved, I would say it is a workaround: [borgbackup/borg#4092](https://github.com/borgbackup/borg/issues/4092)

Do you refer to _vorta_ or borg?

> they suggest that you generate a list and grep on that to imitate a `locate` feature.

This would theoretically allow searching files by name and metadata but not by content. Implementing a fully featured search tool for repository contents within _vorta_ will need a lot of work though.

**Comment by coelner** [comment]
> In vorta you can also mount all archives at once as subfolders of the mount location.

Yes, I wasn't aware that this works and I missed the "mount" button for the complete repository. Only the tooltip shows the difference and I expected the button somewhere else.

I was referring to borg , but I wasn't are that you implement more than borg itself.
```

> https://twitter.com/borgmatic/status/1533662986862092288

vorta list -c user-home.json --find .bashrc --last 5
An instance of Vorta is already running. Opening main window.
```
or did you meant borg?

well, this has side effects. I try this another time. (the buttons for "start backup and cancel" are incorrectly, the backup process is currently active; they need to be switched)
![vorta](https://user-images.githubusercontent.com/16364254/173204192-db9d9b34-b4aa-4c6c-8a92-5609ac46e05c.png)

**Reply by real-yfprojects** [reply]
> > https://twitter.com/borgmatic/status/1533662986862092288
> 
> vorta list -c user-home.json --find .bashrc --last 5
> An instance of Vorta is already running. Opening main window.
> ```
> 
> or did you meant borg?

This twitter post shows a new feature of [borgmatic](https://torsion.org/borgmatic/) not _vorta_.

> well, this has side effects. I try this another time. (the buttons for "start backup and cancel" are incorrectly, the backup process is currently active; they need to be switched) ![vorta](https://user-images.githubusercontent.com/16364254/173204192-db9d9b34-b4aa-4c6c-8a92-5609ac46e05c.png)

Usually this is not the case. Considering opening a new bug report (issue).
